src/app/pages/main/main.component.ts

```diff
@@ -0,0 +1,109 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { Router } from '@angular/router';
+import { Observable, map } from 'rxjs';
+import { of } from 'rxjs/internal/observable/of';
+import { debounceTime } from 'rxjs/operators';
+
+import { ConfirmationModalComponent } from 'src/app/components/confirmation-modal/confirmation-modal.component';
+import { Board } from 'src/app/constants/boardClass';
+import { BoardService } from 'src/app/services/board.service';
+import { ToastService } from 'src/app/services/toast.service';
+
+@Component({
+  selector: 'app-main',
+  templateUrl: './main.component.html',
+  styleUrls: ['./main.component.scss'],
+})
+export class MainComponent implements OnInit {
+  boards$: Observable<Board[]> = new Observable<Board[]>();
+  filteredBoards: Board[] = [];
+  searchTerm: string = '';
+  confirmationModalTitle: string = '';
+  pageSize: number = 10;
+  pageNumber: number = 1;
+  totalPages: number = 1;
+  totalBoards: number = 0;
+  @ViewChild('confirmationModal')
```

Comment:
Please try to follow a single approach to the declaration sequence. I can propose you the following

@decorated properties
public properties
private properties
constructor
lifecycle hooks
public methods
private methods